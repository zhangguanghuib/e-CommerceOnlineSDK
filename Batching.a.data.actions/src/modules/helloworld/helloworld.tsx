/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import * as React from 'react';

import { IHelloworldData } from './helloworld.data';
import { IHelloworldProps } from './helloworld.props.autogenerated';

/**
 *
 * Helloworld component
 * @extends {React.PureComponent<IHelloworldProps<IHelloworldData>>}
 */
class Helloworld extends React.PureComponent<IHelloworldProps<IHelloworldData>> {
    public render(): JSX.Element {
        const { config } = this.props;

        return (
            <div className='row'>
                <div className='col'>
                    <h2>Config Value: {config.title}</h2>
                </div>
            </div>
        );
    }
}

export default Helloworld;
