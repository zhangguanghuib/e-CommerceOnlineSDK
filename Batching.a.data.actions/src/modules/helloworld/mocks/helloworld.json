{
  "id": "R1Module1",
  "config": {
    "title": "Hello world module"
  },
  "data": {
    "actionResponse": {
      "text": "Sample Action Response"
    }
  },
  "typeName": "helloworld"
}