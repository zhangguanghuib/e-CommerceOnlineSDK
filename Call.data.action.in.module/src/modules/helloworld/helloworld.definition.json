{
    "$type": "contentModule",
    "friendlyName": "helloworld",
    "name": "helloworld",
    "description": "Please add meaningful description for authoring",
    "categories": ["helloworld"],
    "tags": ["Additional tags go here"],
    "module": {
        "view": "./helloworld"
    },
    "config": {
        "title": {
            "friendlyName": "Title",
            "description": "example config value",
            "type": "string",
            "default": "Example Config Value",
            "scope": "module",
            "group": "Layout Properties"
        }
    }
}