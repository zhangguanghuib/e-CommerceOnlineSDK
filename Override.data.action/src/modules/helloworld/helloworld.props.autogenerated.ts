/**
 * Copyright (c) Microsoft Corporation
 * All rights reserved. See License.txt in the project root for license information.
 * IHelloworld contentModule Interface Properties
 * THIS FILE IS AUTO-GENERATED - MANUAL MODIFICATIONS WILL BE LOST
 */

import * as Msdyn365 from '@msdyn365-commerce/core';

export interface IHelloworldConfig extends Msdyn365.IModuleConfig {
    title?: string;
}

export interface IHelloworldProps<T> extends Msdyn365.IModule<T> {
    config: IHelloworldConfig;
}
