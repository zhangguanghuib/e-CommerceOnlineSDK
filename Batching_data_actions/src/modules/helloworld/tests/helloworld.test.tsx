/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import { buildMockModuleProps } from '@msdyn365-commerce/core-internal';
/// <reference types="jest" />

// tslint:disable-next-line:no-unused-variable
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Helloworld from '../helloworld';
import { IHelloworldData } from '../helloworld.data';
import { IHelloworldConfig, IHelloworldProps } from '../helloworld.props.autogenerated';

const mockData: IHelloworldData = {
    actionResponse: {
        text: 'Sample Response Data'
    }
};

const mockConfig: IHelloworldConfig = {
    title: 'helloworld'
};

const mockActions = {};

describe('Helloworld', () => {
    let moduleProps: IHelloworldProps<IHelloworldData>;
    beforeAll(() => {
        moduleProps = buildMockModuleProps(mockData, mockActions, mockConfig) as IHelloworldProps<IHelloworldData>;
    });
    it('renders correctly', () => {
        const component: renderer.ReactTestRenderer = renderer.create(<Helloworld {...moduleProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
