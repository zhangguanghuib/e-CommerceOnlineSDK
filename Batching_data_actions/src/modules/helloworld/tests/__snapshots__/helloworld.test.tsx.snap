// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helloworld renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <h2>
      Config Value: 
      helloworld
    </h2>
  </div>
</div>
`;
